### PR TITLE
support for compressed configuration

### PR DESCRIPTION
**Pull request by akshithac-21**
support for compressed configuration

### CODE REVIEW

**Comment by pegaautomationuser** [comment]
Starting PR-672 validation on -> configs

**Comment by pegaautomationuser** [comment]
Starting PR-672 validation on -> configs

**Comment by pegaautomationuser** [comment]
Starting PR-672 validation on -> configs

**Comment by Saurabh-16** [comment]
@akshithac-21 We should also add a documentation for an end user to be able to compress the files in the required format.
It can be added in values.yaml or in read me.

File: charts/pega/values.yaml [hunk 1]
```diff
@@ -338,6 +338,8 @@ global:
         minAvailable: 1
         # maxUnavailable: "50%"
 
+configurations:
+  isCompressed: false
```
Comment (kishorv10) [inline charts/pega/values.yaml:342]: Does it make sense to move this param under global? Also would this be a hidden param or we expose it to clients?

There is already a hidden param for mounting configurations in the global. I guess `.root.Values.global.configurations`. Should we consolidate?

File: charts/pega/templates/pega-environment-config.yaml [hunk 2]
```diff
@@ -43,6 +43,12 @@ data:
   JDBC_TIMEOUT_PROPERTIES_RO: {{ .Values.global.jdbc.readerConnectionTimeoutProperties }}
 {{- else }}
   JDBC_TIMEOUT_PROPERTIES_RO: ""
+{{- end }}
+  # compression flag to decompress the config files of Pega Installation.
+{{- if .Values.configurations.isCompressed }}
+  IS_COMPRESSED: "{{ .Values.configurations.isCompressed }}"
+{{- else }}
+  IS_COMPRESSED: "false"
```
Comment (kishorv10) [inline charts/pega/templates/pega-environment-config.yaml:51]: Variable name looks generic. There is other configuration (jdbc/hazelcast,etc..) in the file. Name does not reflect what configuration being compressed? Can you think of an alternative?

**Comment by taz-pega-work** [comment]
> @akshithac-21 We should also add a documentation for an end user to be able to compress the files in the required format. It can be added in values.yaml or in read me.

thanks @Saurabh-16 !! good call ;-)

**Comment by pegaautomationuser** [comment]
Starting PR-672 validation on -> configs

File: charts/pega/values.yaml (hunk 1 above)
Comment (akshithac-21) [reply charts/pega/values.yaml:348]: moved it under global and this will be exposed to CMC clients.

**Comment by pegaautomationuser** [comment]
Starting PR-672 validation on -> configs

File: charts/pega/templates/pega-environment-config.yaml (hunk 2 above)
Comment (akshithac-21) [reply charts/pega/templates/pega-environment-config.yaml]: yes, Modified the variable name to 'IS_PEGA_CONFIG_COMPRESSED'.

**Comment by pegaautomationuser** [comment]
Starting PR-672 validation on -> configs

**Comment by pegaautomationuser** [comment]
Starting PR-672 validation on -> configs

**Review by MadhuriArugula — APPROVED** [review]

**Review by Saurabh-16 — APPROVED** [review]

**Review by kishorv10 — APPROVED** [review]

**Review by kingakowalska1 — CHANGES_REQUESTED** [review]
Hi, 

I would like to request the following changes to the documentation sections. 

In the readme, please consider changing the text to the following:

### Pega compressed configuration files

To use [Pega configuration files](https://github.com/pegasystems/pega-helm-charts/blob/master/charts/pega/README.md#pega-configuration-files) in compressed format when deploying Pega Platform, replace each file with its compressed format file by completing the following steps:

1. Compress each configuration file using the following command in your local terminal:
```
- cat "<path_to_actual_uncompressed_file_in_local>" | gzip -c | base64 
```
Example for a prconfig.xml file:
```
cat "pega-helm-charts/charts/pega/config/deploy/prconfig.xml" | gzip -c | base64
```
2) Provide the file content with the output of the command for each file executed.
3) Set the `compressedConfigurations` in values.yaml to `true`, as in the following example:
```yaml
  compressedConfigurations: true
```


In the values.yaml file, please consider replacing the comments in lines 124-126 with the following text:

"#" Set the `isCompressed` parameter to `true` when the configuration files under charts/pega/config/deploy are in compressed format.
  "#" For more information, see the “Pega compressed configuration files” section in the Pega Helm chart documentation.

Thanks!
Kinga

**Comment by pegaautomationuser** [comment]
Starting PR-672 validation on -> configs

**Review by kingakowalska1 — APPROVED** [review]
Looks great! Thanks!
Kinga

**Comment by pegaautomationuser** [comment]
Starting PR-672 validation on -> configs

**Pull request merged**